bus/bus.go

```diff
@@ -774,9 +772,15 @@ func (b *Bus) initSettings(ctx context.Context) error {
 			b.logger.Warn(fmt.Sprintf("updated price pinning settings are invalid (%v), they have been overwritten with the default settings", err))
 			updated, _ = json.Marshal(api.DefaultPricePinSettings)
 		}
-
 		if err := b.ss.UpdateSetting(ctx, api.SettingPricePinning, string(updated)); err != nil {
-			return err
+			return fmt.Errorf("failed to update setting '%v': %w", api.SettingPricePinning, err)
+		}
+	} else if pps.Enabled && !b.e.Enabled() {
```

Comment:
Maybe instead of `Enabled` being a field you can make it a method that checks if any value is pinned at all and use that here instead. 
I'd also not update the settings here and instead return a `BadRequest` to make it clear to the user that this is not going to do what they want it to do.

Reply:
This is not executed when the user is updating the settings. These is some compat code in the `bus` that ensures the `bus` is loaded with correct settings. We do return a `400` in the request handler. Since this code is going to be removed entirely when we migrate the settings I'll resolve this comment for now. Feel free to reopen though.